src/main.ts

```diff
@@ -2,28 +2,111 @@ import * as core from '@actions/core'
+import {
+  Inputs,
+  createPullRequest,
+  buildBranchesFromLabels
+} from './github-helper'
 
 const CHERRYPICK_EMPTY =
   'The previous cherry-pick is now empty, possibly due to conflict resolution.'
 
+export type ExecutionStatus = {
+  failed: boolean
+  branch: string
+  inputs?: Inputs
+  msg?: string | undefined | void
+}
+
+export type Statuses = {
+  completedCherryPicks: Array<ExecutionStatus>
+  cherryPickErrors: Array<ExecutionStatus>
+}
+
 export async function run(): Promise<void> {
-  try {
-    const inputs: Inputs = {
-      token: core.getInput('token'),
-      committer: core.getInput('committer'),
-      author: core.getInput('author'),
-      branch: core.getInput('branch'),
-      labels: utils.getInputAsArray('labels'),
-      assignees: utils.getInputAsArray('assignees'),
-      reviewers: utils.getInputAsArray('reviewers'),
-      teamReviewers: utils.getInputAsArray('teamReviewers')
+  const inputs: Inputs = {
+    token: core.getInput('token'),
+    committer: core.getInput('committer'),
+    author: core.getInput('author'),
+    branch: core.getInput('branch'),
+    labels: utils.getInputAsArray('labels'),
+    assignees: utils.getInputAsArray('assignees'),
+    reviewers: utils.getInputAsArray('reviewers'),
+    teamReviewers: utils.getInputAsArray('teamReviewers'),
+    allowUserToSpecifyBranchViaLabel:
+      core.getInput('allowUserToSpecifyBranchViaLabel') || '',
+    labelPatternRequirement: core.getInput('labelPatternRequirement'),
+    userBranchPrefix: core.getInput('userBranchPrefix') || ''
+  }
+
+  const branchesToCherryPick =
+    inputs.allowUserToSpecifyBranchViaLabel === 'true'
+      ? buildBranchesFromLabels(inputs)
+      : [inputs.branch]
+  core.info(`branches to cherry pick ${JSON.stringify(branchesToCherryPick)}`)
+
+  if (!branchesToCherryPick) {
+    core.info(`No branches to cherry pick`)
+    return
+  }
+
+  if (branchesToCherryPick[0] === '') {
+    core.info(`No branches to cherry pick`)
+    return
+  }
+
+  const executions: Array<ExecutionStatus> = []
+
+  core.info(`Executing ${executions.length} cherry picks...`)
+
+  for (const branch of branchesToCherryPick) {
+    try {
+      executions.push(await cherryPickExecution(inputs, branch))
+    } catch (e: any) {
+      executions.push(e)
     }
+  }
+
+  const {completedCherryPicks, cherryPickErrors} = filterExecutionStatuses(
+    executions
+  )
 
-    core.info(`Cherry pick into branch ${inputs.branch}!`)
+  if (completedCherryPicks.length > 0 && cherryPickErrors.length === 0) {
+    core.info(
+      `Finished cherry picking ${JSON.stringify(branchesToCherryPick)}!`
+    )
+  } else if (completedCherryPicks.length > 0 && cherryPickErrors.length > 0) {
+    core.info(`Finished cherry picking ${JSON.stringify(completedCherryPicks)}`)
+    core.info(`Failed to cherry pick ${JSON.stringify(cherryPickErrors)}`)
+  } else {
+    core.info(`Failed to cherry pick ${JSON.stringify(cherryPickErrors)}`)
```

Comment:
```suggestion
    core.info(`Finished cherry picking ${JSON.stringify(branchesToCherryPick)}!`)
    core.info(`Failed to cherry pick ${JSON.stringify(cherryPickErrors)}`)
```
Log CP successes and failures in all cases. No need to if-else.

Reply:
True, I had these blocks throw errors originally. We can condense for now.